diff --git a/limo_base/CMakeLists.txt b/limo_base/CMakeLists.txt
index 3f415b4..9eccae2 100644
--- a/limo_base/CMakeLists.txt
+++ b/limo_base/CMakeLists.txt
@@ -18,6 +18,11 @@ generate_messages(
   std_msgs
 )
 
+catkin_package(
+  INCLUDE_DIRS include
+  LIBRARIES limo_base_node
+  CATKIN_DEPENDS roscpp)
+
 include_directories(
   include
   include/${PROJECT_NAME}
diff --git a/limo_bringup/CMakeLists.txt b/limo_bringup/CMakeLists.txt
index 7473ccf..43bc1a9 100644
--- a/limo_bringup/CMakeLists.txt
+++ b/limo_bringup/CMakeLists.txt
@@ -3,4 +3,8 @@ project(limo_bringup)
 
 find_package(catkin REQUIRED)
 
+catkin_package()
 
+install(DIRECTORY launch param rviz
+  DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
+)
diff --git a/limo_description/CMakeLists.txt b/limo_description/CMakeLists.txt
index e7782ce..458125e 100644
--- a/limo_description/CMakeLists.txt
+++ b/limo_description/CMakeLists.txt
@@ -3,4 +3,8 @@ project(limo_description)
 
 find_package(catkin REQUIRED)
 
+catkin_package()
 
+install(DIRECTORY rviz urdf meshes
+  DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
+)
diff --git a/limo_gazebo_sim/CMakeLists.txt b/limo_gazebo_sim/CMakeLists.txt
--- a/limo_gazebo_sim/CMakeLists.txt
+++ b/limo_gazebo_sim/CMakeLists.txt
@@ -30,8 +30,11 @@ add_library(gazebo_ros_ackerman src/gazebo_ros_ackerman_drive.cpp)
 target_link_libraries(gazebo_ros_ackerman ${catkin_LIBRARIES} ${Boost_LIBRARIES})
 
 install(TARGETS gazebo_ros_ackerman
-ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-RUNTIME DESTINATION ${CATKIN_GLOBAL_BIN_DESTINATION}
+  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+  RUNTIME DESTINATION ${CATKIN_GLOBAL_BIN_DESTINATION}
 )
 
+install(DIRECTORY launch worlds
+  DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
+)
diff --git a/../ydlidar_ros/CMakeLists.txt b/../ydlidar_ros/CMakeLists.txt
--- a/../ydlidar_ros/CMakeLists.txt
+++ b/../ydlidar_ros/CMakeLists.txt
@@ -93,3 +93,7 @@ install(TARGETS ${PROJECT_NAME}_node
 	RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
 	)
 
+install(DIRECTORY launch startup
+	DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
+)
+
